src/lib/rest/rest.cpp

```diff
@@ -193,6 +195,23 @@ static int uriArgumentGet(void* cbDataP, MHD_ValueKind kind, const char* ckey, c
       ciP->uriParamTypes.push_back(val);
     }
   }
+  else if (key == URI_PARAM_Q)
```

Comment:
Note that the Q_URI_PARAM is not used in subscriptions, but in queries. In subscriptions, que q param comes "encapsulated" inside `expression` field (see http://telefonicaid.github.io/fiware-orion/api/v2/ for details). In the case of queries Q_URI_PARAM is used to create an interim Scope (of type FIWARE::StringFilter) which is used in the mongoQueryContext() invocation at mongoBackend.

Thus, for the case of queries, maybe the parsing should be delayed to the scope generation time and the StringFilter object passed as a new attribute in the Scope class, as the current area fields for FIWARE::Location and FIWARE::Location::NGSIv2 filters. Probably the need for a thread var would disapear in this case (and maybe the associated leak), in the same way we haven't needed thread vars for area related stuff.